### ISSUE TITLE
Inconsistent results with same model, same weights, same training data (shuffle off)

### ISSUE THREAD

**Issue by JulietteCarter** [keras-team/keras#7578]
I am loading twice the same model, with the same weights I have saved.
I am training both identical models with the same training data (with shuffle off).
(batch size of 1, 2 iterations for simplicity). 
I would expect the output losses to be exactly the same in both situations. 
Although they do not differ considerable, why are they different??

```
# Model 1 and Model 2 are loaded to be exactly the same
model_1 = load_model(modelPath)
model_1.load_weights(weightsPath)
model_2 = load_model(modelPath)
model_2.load_weights(weightsPath)

#Training Model 1
print('Model 1 results')
model_1.fit(x_train, y_train, batch_size=1, epochs=2, shuffle=False)

#Training Model 2
print('Model 2 results')
model_2.fit(x_train, y_train, batch_size=1, epochs=2, shuffle=False)
```

Model 1 results
Epoch 1/2
1000/1000 [==============================] - 14s - loss: 0.1507    
Epoch 2/2
1000/1000 [==============================] - 14s - loss: 0.1096    
Model 2 results
Epoch 1/2
1000/1000 [==============================] - 14s - loss: 0.1503    
Epoch 2/2
1000/1000 [==============================] - 14s - loss: 0.1093

**Comment by Dref360** [comment]
They differ because most operations are not deterministic by design because of parallelism.
CUDA/cudnn is also non-deterministic by design to speed things up.

**Comment by mahnerak** [comment]
Agree with @Dref360. Had same issue once.
Try to run the same model with CPU.

**Comment by netrapathak** [comment]
I had a problem related to this kind. The same model configuration, with same data would give inconsistent results. Sometimes the training and validation accuracies would increase over epochs, and at times both the accuracies would stay constant. While, both training and validation loss would keep decreasing in both the cases (i.e. while accuracies are constant and even otherwise). I am using parallel convolutions and then merging them before the dense layer. Could someone please explain the reason for this and how to handle such a problem because I have faced this several times with several models where accuracies would remain constant.
